test/mlir/unittests/Analysis/TargetDescriptionAnalysisTest.cpp

```diff
@@ -0,0 +1,81 @@
+//===-- TargetDescriptionAnalysisTest.cpp -----------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include "gc/Analysis/TargetDescriptionAnalysis.h"
+#include "gc/ExecutionEngine/Driver/Driver.h"
+#include "mlir/AsmParser/AsmParser.h"
+#include "mlir/ExecutionEngine/MemRefUtils.h"
+#include "mlir/IR/AsmState.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Pass/PassManager.h"
+#include "llvm/Support/ErrorOr.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/raw_ostream.h"
+#include "gtest/gtest.h"
+#include <memory>
+
+using namespace mlir;
+
+static const char code1[] = R"mlir(
+module attributes {
+dlti.target_system_spec = #dlti.target_system_spec<
+"CPU": #dlti.target_device_spec<
+    #dlti.dl_entry<"L1_cache_size_in_bytes", 49152 : ui32>,
```

Comment:
Currently in `mlir/unittests/Interfaces/DataLayoutInterfacesTest.cpp`, cache sizes are defined as string attributes. Shall we deal with both cases, string & int.